x-pack/apm-server/main.go

```diff
@@ -285,6 +293,33 @@ func newProfilingCollector(args beater.ServerParams) (*profiling.ElasticCollecto
 	return profilingCollector, cleanup, nil
 }
 
+// Fetch the Cluster name from Elasticsearch: Profiling adds it as a field in
+// the host-agent metrics documents for debugging purposes.
+// In Cloud deployments, the Cluster name is set equal to the Cluster ID.
+func queryElasticsearchClusterName(client elasticsearch.Client, logger *logp.Logger) (string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, "/", nil)
```

Comment:
What is the reason for not using client.Info() ?

Reply:
Not sure I follow you: `elasticsearch.Client` here is _not_ the go-elasticsearch/v8 Client, it's an interface with just 2 methods.

Reply:
Not sure I follow you: `elasticsearch.Client` here is _not_ the go-elasticsearch/v8 Client, it's an interface with just 2 methods.